### PR TITLE
Mini challenge 4

### PR DESCRIPTION
**Pull request by vmcruz**
### Description
- Adds a Global provider to keep the state of the application as well as the theme provider.
- Adds a bunch of missing unit tests
---

#### Acceptance Criteria
- [x] The search term is stored and retrieved from the Global Context correctly.
- [x] The appearance theme is stored on the Global Context and applied correctly to the App UI.
- [x] useReducer hook is implemented correctly to manage the Global State.
---

#### Bonus points
- [ ] Testing coverage is above 70%. (Please include a screenshot of the code coverage report in your PR description).
---

#### Coverage screenshots
Screen Shot 2021-03-20 at 4 04 55

### CODE REVIEW

File: src/providers/Global/Global.reducer.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  if (!(action.type in actions)) {
+    throw new Error(`${action.type} is not a valid action for Global.reducer`);
+  }
+
+  const actionCallee = actions[action.type];
+
+  return actionCallee(state, action.payload);
+}
+
+export { reducer, initialState };
```
Comment (PacoMojica) [inline src/providers/Global/Global.reducer.js:58]: 👌 I like this approach, it's way cleaner than using a switch with strings

**Review by PacoMojica — COMMENTED** [review]
Hello Victor 👋, you challenge is really really good. I see that the test coverage is around 65%, even the branch percentage. I think you have the best tests among the challenges I have reviewed, they are well defined and have meaningful edge cases. I like the approach you have with the Global provider and the global reducer is 👌. I'm glad you are part of the team in the external bootcamp. Great work 👍👍
